generic_config_updater/patch_sorter.py

```diff
@@ -670,6 +759,28 @@ def _validate_table(self, table, config):
         # the only invalid case is if table exists and is empty
         return table not in config or config[table]
 
+class PortCriticalMoveValidator:
```

Comment:
>PortCriticalMoveValidator [](http://example.com/codeflow?start=6&length=25)

I believe there will more cases other than port in future. The concept is similar to a lock. When a node is locked, the manipulation of the node subtree is banned.

Can we add a yang model extension to capture this behavior, and apply the extension to port as a first case?

Reply:
The idea of a locking a node is interesting. My thought was to name it `TransitionalValueValidator` later if we need to support more cases, and also move it to YANG. Added that as a TODO in line 390.

>     # TODO: port-critical fields are hard-coded for now, it should be moved to YANG models
>     #  NOTE: PortCritical validation logic can be generalized to be validation for any config that
>     #        requires another config to be of a specific value. Call the validator "TransitionalValueValidator"
>     #        For the case of port-critical validation the transitional value would be "admin_status=down".

Let me think about the lock idea.

Reply:
Sorry took me a bit to think about this, had other work items.

I think the idea of lock is similar to what we have today for the validation. If the validation fails we cannot do the update.

`The concept is similar to a lock. When a node is locked, the manipulation of the node subtree is banned`, I think this is the same as what we have for validation `if validation of a move that is changing a specific part of the config fails, the move is rejected.`

Reply:
Updated to handle any config and named `RequiredValueValidator`